web/controllers/rules/rules.ex

```diff
@@ -2,32 +2,29 @@ defmodule Dwylbot.Rules do
   @moduledoc """
   Functions to check the wrokflow contribution rules
   """
-  def check(payload, author) do
-    assignees = payload["assignees"]
-    labels = payload["labels"]
-    in_progress = Enum.any?(labels, fn(l) -> l["name"] == "in-progress" end)
-    if in_progress && Enum.empty?(assignees) do
-      [%{error_type: "inprogress_noassignees", author: author}]
-    else
-      []
-    end
+  alias Dwylbot.Rules.List, as: RulesList
+
+  def apply_and_check_errors(payload) do
+    rules = RulesList.get_rules()
+    rules
+    |> Enum.filter(fn(m) -> apply(m, :apply?, [payload]) end)
+    |> Enum.map(fn(m) -> apply(m, :check, [payload]) end)
+    |> Enum.filter(fn(e) -> e != nil end)
+  end
```

Comment:
This is fun, might be good to have a bit of `@doc`umentation showing what a typical `payload` looks like.
I think you should explain here what a typical `apply?` and `check` function should look like here and do.
It may be a little confusing piping `rules` into something which then has the argument of `m` (which I think means `Module`)
I think keep the list naming consistent as either `rules` or `modules`